black_it/loss_functions/base.py

```diff
@@ -66,12 +76,39 @@ def compute_loss(
             )
             weights = self.coordinate_weights
 
+        if self.coordinate_filters is None:
+            # a list of identity functions
+            filters = [None] * num_coords
+        else:
+            nb_coordinate_filters = len(self.coordinate_filters)
+            assert_(
+                nb_coordinate_filters == num_coords,
+                (
+                    "the length of coordinate_filters should be equal "
+                    f"to the number of coordinates, got {nb_coordinate_filters} and {num_coords}"
+                ),
+                exc_cls=ValueError,
+            )
```

Comment:
I guess this validation could be moved into the constructor; otherwise an error would occur only when `compute_loss` is called, which could surprise the user a bit

Reply:
We discussed this already. Unfortunately it is not possible since the BaseLoss constructor does not know the number of coordinates when it's called

Reply:
right, but a sanity check with `coordinate_weights` could be useful